### PR TITLE
labels: add version labels

### PR DESCRIPTION
**Pull request by mscdex**

### CODE REVIEW

**Comment by mscdex** [comment]
I wasn't sure exactly how to test the webhook, but going by the github webhook docs that *should* be the correct property for obtaining the base branch.

**Comment by phillipj** [comment]
> I wasn't sure exactly how to test the webhook ...

Np, no one else has gotten around creating automated tests for webhooks so far - hopefully some day :) The easiest way to test webhooks locally ATM would be via SSE relay functionality.

@williamkapke could you add the necessary relay webhook to https://github.com/TestOrgPleaseIgnore/node?

**Comment by jbergstroem** [comment]
Perhaps just mock the api response from the version we're talking to?

**Comment by phillipj** [comment]
Yeah, I've got a branch ready to push soon which mocks every call made to GH

**Comment by jbergstroem** [comment]
Awesome!

File: lib/node-labels.js [hunk 1]
```diff
@@ -53,12 +53,27 @@ const exclusiveLabelsMap = new Map([
...
+    }
+    baseBranch = ''
+  }
+
+  const labels = (exclusiveLabels.length > 0)
+                  ? exclusiveLabels
+                  : matchAllSubSystem(filepathsChanged, limitLib)
+
+  // Add version labels if PR is made against a version branch
+  const m = /^(v\d+\.(?:\d+|x))(?:-|$)/.exec(baseBranch)
```
Comment (Fishrock123) [inline lib/node-labels.js]: I would prefer to check `-staging$` directly in the last one.

**Comment by phillipj** [comment]
Landed in 3a30e94f4c6381df2e79cbef828826e311f2cf86

So sorry for stalling this for so long @mscdex! Included an integration test ensuring the `v6.x`  got added when the webhook payload said the PR targeted `v6.x-staging`.

**Comment by phillipj** [comment]
This is now live in production

**Comment by Fishrock123** [comment]
Comment was ignored? :/

**Comment by phillipj** [comment]
> Comment was ignored? :/

@Fishrock123 sry, didn't see your comment! I'll get a PR up right away... Was this a coincident you having a look at this old & stalled PR, and me working to get it merged??

**Comment by mscdex** [comment]
Hooray, it works!: https://github.com/nodejs/node/pull/8437

**Comment by phillipj** [comment]
Yiiiiiha, nice work!

@mscdex you got any objections to this
https://github.com/nodejs/github-bot/pull/70?

**Pull request closed without merging**